### DISCUSSION TITLE
关于时间段划分与（可能的）新功能的小建议

### DISCUSSION THREAD

**Discussion by YanDao0313** [Class-Widgets/Class-Widgets#50] [建议]
软件很好看！完成度也很高！~~单冲外观这一点就让我下定决心把我班用的`ClassIsland`迁移到这边来了（不是~~  
咳咳，切入正题，就目前的情况而言，个人认为我想分享以下几点建议（大致按重要程度倒序）：  

1. 希望课表时间段除上/下午之外增加**晚自习时间段**
    - 其实大家都知道要上晚自习的吧（逃
    - 其实现有版本也还可以了，但是全部集中至下午会导致整体易读性的下降、不适合交接工作
2. 关于课表编辑界面
    - 平心而论，`ClassIsland`的易读性会比目前的`ClassWidgets`更好，希望可以参考  
       ![image](https://github.com/user-attachments/assets/0a151b51-a144-4b8f-945b-fb575ec78ca5)  
       ![image](https://github.com/user-attachments/assets/98520ef6-9b23-4b89-83d8-aa7aa76cbc2e)  
    - 其实也没必要弄得太麻烦，现有基础上其实修改一下排版（增加时间信息，分栏，分割线，etc.）也能有一个比较好的提升  
      ![39e2f0717433d1693beaab0addd5c02f](https://github.com/user-attachments/assets/ffd010d2-b921-485a-a9b8-76fc7e385d01)  
    - 另外就是修改课程安排的操作逻辑，其实也可以做成像目前“课表预览”板块一样的设计，会比现在的更方便一些
    - 我看到issue区有关于“多段课间休息”的建议，其实有一个“曲线救国”的方法——允许重命名“课间休息”时段的播报名称（bushi
3. 希望增加**天气预报与气象预警功能**
    - ~~这是一个**巩固你电教委地位**的好功能！~~    
    - 具体实现可以参考`ClassIsland`~~，当然也不一定用小米天气~~   
    - 不一定需要具备像`ClassIsland`一样细致的调控，准许在一部分情况下开启即可（如给课间休息打上属性，则在课间休息时播报未来天气、气象预警）  
    - 其实如果可以在自定义中允许用户将“日期”处修改成实时天气+预警图标也很好  
4. 在开启单双周课表的情况下，若只设置了单周/双周的课表，则另一周**自动复用已设置的课表**（现在需要手动复制，不过也不算大问题啦~）  
5. 不清楚现在是怎么安排的隐藏逻辑，在测试时经常发现它自己隐藏了然后就出不来了（还不好复现（？  
6. 允许用户**开启或关闭提示音**，引入类似于ClassIsland的更为醒目的**全屏提醒动画**（尤其是考试场景下......）
8. 可以引入**室内外课程的区分**（室外课程**提前提醒**），便于提前提醒认真内卷的同学  ~~以免体育老师发飙~~  
9. 可以引入**教师名称播报机制**（下节课是：生物  xx老师  07:21~11:45）  ~~适合虚荣心强的班主任~~
10. 考虑修改部分软件内文本
    - 例如“当前活动”可以改为“现在是” / “本节课是”  
      ![image](https://github.com/user-attachments/assets/0062857e-3b6d-41f0-9cca-834f35b98e53)
    - 此外，提醒中的文本也建议改成对应下一节点（即将开始的节点）的内容，而不是展示上一节点（结束的节点）的内容
11. 在“自定义”中修改内容之后，界面似乎不会即时更新，这会比较影响使用体验  
12. 考虑引入**时间逐日自动偏移**，适合不会定期校时的学校（比如我校
    - 即每天在原有基础上自动增加/减少一定时间偏移量
13. 考虑引入**教学安全模式**，如遇崩溃后无须人工介入即可自动退出/重启~~，以免公开课上出现尴尬场面~~
14. 考虑进一步开放自定义，**增加多种小组件**（文本、实时天气等等）
15. 考虑允许当前活动倒计时**自定义倒计时粒度**（分钟，或者显示当前节点的起止时间, etc.）
      ![image](https://github.com/user-attachments/assets/8412cb99-7b41-45dc-8e83-4dd978bde969)
16. 软件的主题色是如何决定出来的呢？希望可以**允许用户自定义主题色**（人为指定/从壁纸中提取/沿用Windows主题色, etc.）
17. 建立软件**使用文档**，进一步降低上手门槛
18. 完善**触屏兼容性**

当然，以上只是本人**十分浅薄**的建议，也**只是提供一些灵感**（软件本身已经**很好了**）  
另外，似乎UI界面里面没能正常渲染粗体（逃  
![image](https://github.com/user-attachments/assets/c89aa539-7603-41c8-b872-1e5f9dc70a84)

**Comment by jiangyin14** [comment]
ci人震怒（😡）

**Comment by DryIce-cc** [comment]
ci 人就爱看这种一语中的（😆）

**Comment by 6JinQuan** [comment]
好建议！！！说出了本电教的心声！！！！
